internal/unionstore/memdb.go

```diff
@@ -833,3 +833,13 @@ func (n *memdbNode) getKeyFlags() kv.KeyFlags {
 func (n *memdbNode) setKeyFlags(f kv.KeyFlags) {
 	n.flags = (^nodeFlagsMask & n.flags) | uint16(f)
 }
+
+// UnsafeRemoveRecord removes a record from the mem buffer. It should be only used for test.
+func (db *MemDB) UnsafeRemoveRecord(key []byte) {
```

Comment:
After some consideration, I think `RemoveFromBuffer` should be a more clear and understandable name.
By the way, you can remove the `DeleteKey` function in memdb_test.go and change its usages to this new one.